patcher/src/ts/redux/modules/servers.ts

```diff
@@ -180,14 +194,12 @@ export default function reducer(state: ServersState = initialState, action: any
       return Object.assign({}, state, {
         currentServer: action.server
       });
-    case UPDATE_SERVER:
-      let servers = state.servers;
-      const index = servers.findIndex((s: Server) => s.name == action.server.name);
-      if (index > -1) servers[index] = action.server;
-      else servers.push(action.server);
+    case UPDATE_SERVER_POPULATION:
+      state.serverPopulations[action.serverPop.serverName] = action.serverPop;
```

Comment:
Never update the current state object directly.  Nor should you modify function parameters, which this is both in this case.

Instead clone the array:

```
let serverPopulations = state.serverPopulations.slice();
serverPopulations[action.serverPop.serverName] = action.serverPop;
```